Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/XLGppDataStore.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.ContractsLight;
+using BuildXL.Engine.Cache.KeyValueStores;
+using BuildXL.Execution.Analyzer;
+using BuildXL.Scheduler.Tracing;
+using BuildXL.Utilities;
+using Google.Protobuf;
+
+namespace BuildXL.Analyzers.Core.XLGPlusPlus
+{
+    public class XLGppDataStore
+    {
+        /// <summary>
+        /// Rocks DB Accessor for XLG++ data
+        /// </summary>
+        public KeyValueStoreAccessor Accessor { get; set; }
+
+        public XLGppDataStore()
+        {
+
+        }
+
+        /// <summary>
+        /// Open the datastore and populate the KeyValueStoreAccessor for the XLG++ DB
+        /// </summary>
+        /// <returns>Boolean if datastore was opened successfully</returns>
+        public bool OpenDatastore(string storeDirectory,

```

Comment:
>OpenDatastore [](start = 20, length = 13)

what do you think of moving this into the ctor? the class is not usable until you call into this method.

Reply:
Now that it's been moved to the constructor I think you can remove this method

---
In reply to: [306048177](https://github.com/microsoft/BuildXL/pull/636#discussion_r306048177) [](ancestors = 306048177)